src/archival/utils.rs

```diff
@@ -59,6 +74,50 @@ pub async fn is_row_exists(pool: &PgPool, row_id: i32) -> bool {
     }
 }
 
+pub async fn make_archival_network_request(
+    url: &str,
+    endpoint_url: &str,
+) -> Result<ArchivalResponse, ArchivalError> {
+    let settings = Settings::new().expect("Config settings are not configured properly");
+    let mut headers = header::HeaderMap::new();
+    headers.insert("Accept", "application/json".parse().unwrap());
+    headers.insert(
+        "Authorization",
+        format!(
+            "LOW {}:{}",
+            settings.wayback_machine_api.myaccesskey, settings.wayback_machine_api.mysecret
+        )
+        .parse()
+        .unwrap(),
+    );
+    headers.insert(
+        "Content-Type",
+        "application/x-www-form-urlencoded".parse().unwrap(),
+    );
+
+    let client = Client::builder()
+        .redirect(reqwest::redirect::Policy::none())
+        .build()
+        .unwrap();
```

Comment:
You shouldn't create a reqwest client each time you need to do a request. You can do a global client that get reused each time with `once_cell`'s `Lazy`. Exemple [here](https://github.com/RustyNova016/listenbrainz-cli-tools/blob/99200b1838f37e61d3f41645dface968f06a194e/src/models/data/musicbrainz_database.rs#L25C1-L26C60)

Reply:
Implemented, thanks for this!